### ISSUE TITLE
Make proxyEnabled configuration thread-specific

### ISSUE THREAD

**Issue by doctordrue** [selenide/selenide#2743]
## The problem
Configuration.proxyEnabled is global so either all tests in a will create WebDriver with selenide proxy or none of them.

## Details
Currently because of Configuation is global calling `Configuration.proxyEnabled=true` in configuration methods (e.g. @BeforeMethod in TestNG) & 'Configuration.proxyEnabled=true' in tear down (e.g. @AfterMethod) affects other test if running in parallel.
It would be great to have at least thread-specific configuration for proxy, so enabling proxy in one thread will not enable it in another.

## Tell us about your environment
* **Selenide Version**: 7.0.2

**Comment by asolntsev** [comment]
Duplicate of https://github.com/selenide/selenide/issues/1372
